software/controller/lib/hal/i2c.cpp

```diff
@@ -282,9 +271,9 @@ void STM32Channel::Init(I2CReg *i2c, DmaReg *dma, Speed speed) {
   i2c_->timing.full_reg = static_cast<uint32_t>(speed);
 
   // Setup DMA channels
-  if (dma != nullptr) {
-    SetupDMAChannels(dma);
-  }
+  //  if (dma != nullptr) {
+  SetupDMAChannels(dma);
```

Comment:
Yeah, that should work fine since `SetupDMAChannels` called with nullptr will never match `if (dma == map.dma_base && i2c_ == map.i2c_base)`.
Why leave it as a comment?

Reply:
Note that it cannot be `nullptr` now because it is a n `Enum` which can only be one of the 2 DMAs. This is all slightly in transition and once we refactor and abstract some of the lower level interfaces, this class will need a bit of a makeover.